src/open-api/from-open-api.ts

```diff
@@ -11,6 +11,12 @@ const supportedHttpMethods = Object.keys(
   http,
 ) as unknown as SupportedHttpMethods
 
+export type MapOperationFunction = (
+  url: string,
```

Comment:
I recommend using object-as-argument pattern:

```ts
export type X = (args: {
  url: string,
  method: Y,
  operation: Z
}) => T
```

We should've done that in `fromTraffic` too, and that's something I'd like to do in the future versions.

Reply:
I am curious if we want to pass the PathItemObject instead and then let the user return the responses it wants to map?

Reply:
@weyert, I'm not sure whether that's beneficial. We may think more about how to decide which responses from the spec to pick. Right now, the library recommends using a search parameter but that means modifying the system's behavior, which I don't like. 

You can use this map function instead... I'm thinking if that has any complications. For once, if you drastically transform the spec, you won't be able to know what is actually the source of truth for the network. It kind of becomes your map function, which is bad.

Reply:
It would be great to rely on something like response scenarios name/ids. So you cannot modify them, maintaining the source of truth, but can reference/swap per your need.

Reply:
changed to object argument: c49d3c72f9b64a5094faeef752cca7a097d36210

Reply:
@kettanaito Sorry, what do you mean with system's behaviour? Some other mocking solutions allow you to set a HTTP header to define which example of the responses should be returned.

Reply:
@weyert, I meant the server you are describing in your OAS document. Basically, my main argument is to support filtering primarily, since the more your runtime modifies your spec, the least useful your spec becomes. If you need to modify an operation in your spec, that's an indicator your spec isn't covering all use cases (otherwise, why would you want to modify it?) I'd rather recommend developers extend their specs to cover the scenarios they want, and have Source provide them with the means to target specific scenarios in test (like we have with triggering particular response examples by the status code search parameter).

Reply:
That's true. The idea of this PR is to limit the number of endpoints get mocked. If you have a specification with hundreds of endpoints then you might not need to mock all of them. So I like that the idea of this PR to only mock a subset of the endpoints. As you said I can also imagine that you might want to only mock a particular response example of an endpoint. Both are valid use cases but I agree with you that just remapping endpoints shouldn't be an end goal here.

Reply:
But I do agree that it would be nice to cherry-pick certain responses (see #55).